File: src/main/java/com/mini/credit/dto/membre/MembreResponse.java
package com.mini.credit.dto.membre;

import com.mini.credit.enums.Sexe;
import com.mini.credit.enums.StatutMembre;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
public class MembreResponse {

    private Long id;

    private String codeMembre;

    private String nom;
    private String postnom;
    private String prenom;
    private String nomComplet;

    private Sexe sexe;
    private LocalDate dateNaissance;

    private String telephonePrincipal;
    private String telephoneSecondaire;
    private String email;

    private String adresse;
    private String quartier;
    private String commune;
    private String ville;

    private String professionActivite;
    private String lieuActivite;
    private String sourceInscription;

    private Long siteId;
    private String siteNom;

    private Long agentId;
    private String agentMatricule;

    private LocalDate dateAdhesion;
    private StatutMembre statut;

    private String photoUrl;
    private String observation;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/app/features/membres/models/membre-response.ts
import { Sexe } from "../../../shared/enums/sexe.enum";
import { StatutMembre } from "../enum/statut-membre";



export interface MembreResponse {
  id: number;

  codeMembre: string;

  nom: string;
  postnom?: string;
  prenom?: string;
  nomComplet?: string;

  sexe?: Sexe;
  dateNaissance?: string;

  telephonePrincipal?: string;
  telephoneSecondaire?: string;
  email?: string;

  adresse?: string;
  quartier?: string;
  commune?: string;
  ville?: string;

  professionActivite?: string;
  lieuActivite?: string;
  sourceInscription?: string;

  siteId: number;
  siteNom: string;

  agentId?: number;
  agentMatricule?: string;

  dateAdhesion: string;
  statut: StatutMembre;

  photoUrl?: string;
  observation?: string;

  createdAt: string;
  updatedAt: string;
}